### PR TITLE
[dev] Apply pending python3 patches to acceptance tests

### PR DESCRIPTION
**Pull request by achilleasa**
When 2.9 was forward ported to develop, some of the python3 upgrades for the acceptance tests were not merged successfully due to the change from `juju run` (which now runs actions) to `juju exec`.

This PR ensures that everything is consistent.

### CODE REVIEW

File: acceptancetests/jujupy/client.py
Comment (achilleasa) [inline acceptancetests/jujupy/client.py:2622]: copied over from acceptancetest/utility.py

File: acceptancetests/jujupy/client.py
Comment (achilleasa) [inline acceptancetests/jujupy/client.py:1336]: I made a standalone method for rebooting machines and also plan on back-porting it to 2.9 so we can use the same API instead of `juju ssh` (develop) and `juju run` (2.9)

File: acceptancetests/deploy_stack.py
Comment (achilleasa) [inline acceptancetests/deploy_stack.py:205]: `remote.cat` decodes the bytes it gets back internally so contents is already a string and needs no decoding.

**Comment by achilleasa** [comment]
Please do not merge this. It still needs changes.

**Review by SimonRichardson — APPROVED** [review]
I wonder what else we've missed?

**Comment by achilleasa** [comment]
> I wonder what else we've missed?

It's probably a matter of merging, waiting for the tests to run and fix the broken ones just as we did with the 2.9 branch. Most of them were choking in the get_token method so they should hopefully be mostly passing after this PR lands.

**Comment by achilleasa** [comment]
$$merge$$

**Pull request merged**
